### PR TITLE
ViT Integration

### PR DESCRIPTION
**Pull request by KSGulin**
- Add: setup_integration.sh
- Update: Readme.md
- Remove: train.py

### CODE REVIEW

File: integrations/rwightman-timm/README.md [hunk 1]
```diff
@@ -23,64 +23,51 @@ Using this integration, you will be able to apply SparseML optimizations
...
+Recipes encode the instructions and hyperparameters for sparsifying a model using modifiers to the training process.
+The modifiers can range from pruning and quantization to learning rate and weight decay.
+When appropriately combined, it becomes possible to create highly sparse and accurate models.
 
-* `--sparseml-recipe` is a required parameter
-* `--epochs` will now be overridden by the epochs set in the SparseML recipe
-* Modifiers will log their outputs to the console as well as to a tensorboard file
-* After training is complete, the final model will be exported to ONNX using SparseML
+This integration adds a `--recipe` argument to the 
+[`train.py` script](https://github.com/neuralmagic/yolact/blob/master/train.py).
```
Comment (rahul-tuli) [inline integrations/rwightman-timm/README.md:58]:
```suggestion
[`train.py` script](https://github.com/neuralmagic/pytorch-image-models/blob/master/train.py).
```

File: integrations/rwightman-timm/README.md [hunk 2]
```diff
@@ -23,64 +23,51 @@ Using this integration, you will be able to apply SparseML optimizations
...
+The argument loads an appropriate recipe while preserving the rest of the training pipeline.
+Popular recipes used with this argument are found in the [`recipes` folder](./recipes).
```
Comment (rahul-tuli) [inline integrations/rwightman-timm/README.md:60]: No recipes folder in the repo yet

File: integrations/rwightman-timm/README.md [hunk 3]
```diff
@@ -94,30 +81,36 @@ python integrations/timm/train.py \
...
+The latest commit hash that `train.py` is based on is included in the docstring.
+
+
+The following table lays out the root-level files and folders along with a description for each.
 
-Training from a SparseZoo recipe and checkpoint with sparse transfer learning enabled
+| Folder/File Name     | Description                                                                                                           |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------|
+| recipes              | Typical recipes for sparsifying ViT models along with any downloaded recipes from the SparseZoo (WIP).                      |
+| tutorials            | Tutorial walkthroughs for how to sparsify ViT models using recipes (WIP).                                                   |          |
```
Comment (rahul-tuli) [inline integrations/rwightman-timm/README.md:94]: making a note here, as a reminder to push out the recipes and tutorial

File: integrations/rwightman-timm/README.md [hunk 4]
```diff
@@ -94,30 +81,36 @@ python integrations/timm/train.py \
...
+| README.md            | Readme file.                                                                                                        |
+| setup_integration.sh | Setup file for the integration run from the command line.   
+
+### Exporting for Inference
+
+After sparsifying a model, the `export.py` script can be run to convert the model into an [ONNX](https://onnx.ai/) deployment format.
```
Comment (rahul-tuli) [inline integrations/rwightman-timm/README.md:100]: Add a link to `export.py`

**Review by rahul-tuli — CHANGES_REQUESTED** [review]

**Comment by rahul-tuli** [comment]
Looks good to me overall, minor nits; I didn't take a super close looks at the training and export commands though

File: integrations/rwightman-timm/README.md [hunk 5]
```diff
@@ -23,64 +23,51 @@ Using this integration, you will be able to apply SparseML optimizations
...
 
-Documentation on the original script can be found
-[here](https://rwightman.github.io/pytorch-image-models/scripts/).
-The latest commit hash that `train.py` is based on is included in the docstring.
-
+ViT recipes are not yet available on SparseZoo, but are in progress.
 
-#### Example Command
-Training from a local recipe and checkpoint
+Using a local recipe and checkpoint, a pruning a model can be done by running the following command from within the root of this integration's folder
```
Comment (markurtz) [inline integrations/rwightman-timm/README.md:70]:
```suggestion
Using a local recipe and checkpoint, pruning a model can be done by running the following command from within the root of this integration's folder
```

File: integrations/rwightman-timm/setup_integration.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Integration setup command to setup the folder so it is ready to train and sparsify models.
+# Creates a pytorch-image-models folder next to this script with all required dependencies from the rwightman/pytorch-image-models repository.
+# Command: `bash setup_integration.sh`
+
+git clone https://github.com/neuralmagic/pytorch-image-models.git
+cd pytorch-image-models
```
Comment (markurtz) [inline integrations/rwightman-timm/setup_integration.sh:8]: @KSGulin we should add a release branch on the timm repo and change to that branch here so we can guarantee ongoing changes and improvements won't affect the latest stable release

**Review by markurtz — CHANGES_REQUESTED** [review]

File: integrations/rwightman-timm/README.md [hunk 7]
```diff
@@ -23,64 +23,51 @@ Using this integration, you will be able to apply SparseML optimizations
...
-i.e. `zoo:cv-classification/resnet_v1-50/pytorch-rwightman/imagenet-augmented/pruned_quant-aggressive?recipe_type=transfer_learn`.
-This will run a recipe that holds the optimized sparsity structure the same while allowing
-non-zero weights to be updated during training, so pre-learned optimizations can be applied
-to different datasets.
-
-To load the base weights for a SparseZoo recipe as the initial checkpoint, set
-`--initial-checkpoint` to `zoo`.  To use the weights of a SparseZoo model as the
-initial checkpoint, pass that model's SparseZoo stub prefixed by `zoo:` to the
-`--initial-checkpoint` argument.
+The `setup_integration.sh` file will clone the pytorch-image-models repository with the SparseML integration as a subfolder.
```
Comment (rahul-tuli) [inline integrations/rwightman-timm/README.md:46]: nit: link the repo
```suggestion
The `setup_integration.sh` file will clone the [pytorch-image-models](https://github.com/neuralmagic/pytorch-image-models.git) repository with the SparseML integration as a subfolder.
```

Can also put an anchor at the end of the file ass follows:
```
[pytorch-image-models]: https://github.com/neuralmagic/pytorch-image-models.git
```
and then use `[pytorch-image-models]` everywhere

File: integrations/rwightman-timm/setup_integration.sh [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+pip install -r requirements.txt
```
Comment (rahul-tuli) [inline integrations/rwightman-timm/setup_integration.sh:9]: nit: newline

**Review by rahul-tuli — DISMISSED** [review]

**Review by bfineran — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**